testing_mad.ipynb

```diff
@@ -14,237 +14,2049 @@
+       "      <td>12.3415</td>\n",
+       "      <td>107.6772</td>\n",
+       "      <td>9.1659</td>\n",
+       "      <td>66.6025</td>\n",
+       "      <td>212.2431</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "<p>5 rows × 1332 columns</p>\n",
+       "</div>"
+      ],
       "text/plain": [
-       "0    1.0\n",
-       "1    3.0\n",
-       "2    0.0\n",
-       "3    5.0\n",
-       "4   -3.0\n",
-       "dtype: float64"
+       "                   gene_id             gene gene_symbol  GSM2667747  \\\n",
+       "0     ENSG00000153037SRP19  ENSG00000153037       SRP19    894.5558   \n",
+       "1    ENSG00000155115GTF3C6  ENSG00000155115      GTF3C6    851.6143   \n",
+       "2   ENSG00000184083FAM120C  ENSG00000184083     FAM120C    632.3881   \n",
+       "3    ENSG00000008196TFAP2B  ENSG00000008196      TFAP2B     23.2873   \n",
+       "4  ENSG00000234754C1orf140  ENSG00000234754    C1orf140    120.9504   \n",
+       "\n",
+       "   GSM2667748  GSM2667749  GSM2667750  GSM2667751  GSM2667752  GSM2667753  \\\n",
+       "0    357.1081    187.4946   1421.0479    456.0143    802.6670    124.8560   \n",
+       "1    187.0203    243.4404   2046.5828    476.0532   1977.8905     94.7822   \n",
+       "2    285.1058    122.8606    832.8702    957.4550   1495.5920    184.7934   \n",
+       "3      2.0018     19.1983     54.7994     19.0940    974.5680     88.7088   \n",
+       "4    135.9405      8.7069     43.7292     80.6176     75.2029     60.4073   \n",
+       "\n",
+       "   ...  GSM2692589  GSM2692590  GSM2692591  GSM2692592  GSM2692593  \\\n",
+       "0  ...   3577.9142   3429.3843   3143.5853   4008.3218   5379.2540   \n",
+       "1  ...   2675.3643   2541.6896   2230.9446   2901.2439   3445.3752   \n",
+       "2  ...    447.2578    402.4417    404.5406    501.3679    928.1712   \n",
+       "3  ...      0.0000      0.0000      0.0000      0.0000      0.0000   \n",
+       "4  ...     40.5659     33.4383     20.8078      6.3592      9.1679   \n",
+       "\n",
+       "   GSM2692594  GSM2693346  GSM2693347  GSM2693348  GSM2693349  \n",
+       "0   5677.6213    305.5665     16.3645     36.9488    166.5698  \n",
+       "1   3716.4132    188.6653     41.2565    102.6781    594.5893  \n",
+       "2   1065.6067    359.6045    110.8197   2788.2711    417.6723  \n",
+       "3      0.0000      0.0000      0.0000      0.9970      0.0000  \n",
+       "4     12.3415    107.6772      9.1659     66.6025    212.2431  \n",
+       "\n",
+       "[5 rows x 1332 columns]"
       ]
      },
-     "execution_count": 15,
+     "execution_count": 4,
      "metadata": {},
      "output_type": "execute_result"
     }
    ],
    "source": [
-    "# calculating the median of the data frame\n",
-    "df_median = df.median(axis='columns', numeric_only=True)\n",
-    "df_median"
+    "# add gene_id as a column in dataframe\n",
+    "df_4['gene_id'] = df_4['gene'] + df_4['gene_symbol']\n",
```

Comment:
```suggestion
    "df_4['gene_id'] = df_4['gene'] + "_" + df_4['gene_symbol']\n",
```
In order to recover this info, I think it's better to have a delimiter between the two fields. However, because your function shouldn't change anything, you should be able to just concatenate this information back together, so I'm not sure you actually need to do this!

Reply:
I concatenated gene and gene _symbol here because I can only set an index to an existing column couldn't think of any other way to ensure that the gene_id consisted of both the gene and the gene_id